### PR TITLE
fix(iam-sqs): Handle exceptions for non-existent resources

### PR DESCRIPTION
**Pull request by jfagoagas**
### Description

- SQS: Handle `AWS.SimpleQueueService.NonExistentQueue` errors
- IAM: Handle `NoSuchEntity` and warnings. I've changed `NoSuchEntityException` to `NoSuchEntity` since the `error.response["Error"]["Code"]` contains just the `NoSuchEntity`.

The IAM change can be tested with this code:
```python
from boto3 import client
from botocore.client import ClientError

iam_client = client("iam", "eu-west-1")
arn = "arn:aws:iam::123456789012:policy/test-policy"

try:
    iam_client.list_policy_tags(PolicyArn=arn)
except ClientError as error:
    print(error.response["Error"]["Code"])
```

### License

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: prowler/providers/aws/services/iam/iam_service.py [hunk 1]
```diff
@@ -697,7 +697,7 @@ def __get_last_accessed_services__(self):
                     ]
 
                 except ClientError as error:
-                    logger.error(
+                    logger.warning(
```
Comment (MrCloudSec) [inline prowler/providers/aws/services/iam/iam_service.py]: why did you change this one?

File: prowler/providers/aws/services/iam/iam_service.py [hunk 2]
```diff
@@ -717,7 +717,7 @@ def __get_access_keys_metadata__(self):
                             "AccessKeyMetadata"
                         ]
                 except ClientError as error:
-                    logger.error(
+                    logger.warning(
```
Comment (MrCloudSec) [inline prowler/providers/aws/services/iam/iam_service.py]: and this one?

File: prowler/providers/aws/services/iam/iam_service.py (hunk 1 above)
Comment (jfagoagas) [reply prowler/providers/aws/services/iam/iam_service.py]: Good catch, I've included both exceptions.

File: prowler/providers/aws/services/iam/iam_service.py (hunk 2 above)
Comment (jfagoagas) [reply prowler/providers/aws/services/iam/iam_service.py]: Good catch, I've included both exceptions.

**Review by MrCloudSec — APPROVED** [review]
